src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -232,14 +220,14 @@ public T GetValue(CancellationToken cancellationToken)
         }
         else
         {
-            Contract.ThrowIfNull(_synchronousComputeFunction);
+            Contract.ThrowIfFalse(HasSynchronousComputeFunction);
 
             T result;
 
             // We are the active computation, so let's go ahead and compute.
             try
             {
-                result = _synchronousComputeFunction(cancellationToken);
+                result = InvokeComputeFunction(cancellationToken);
             }
             catch (OperationCanceledException)
```

Comment:
This is incorrectly written, but too risky to change as part of this pull request. This `catch` block should have had a `when (cancellationToken.IsCancellationRequested)` clause.